### PR TITLE
Improve admin_default navbar & responsiveness

### LINKED ISSUE #710: Unable to view and use of Admin Panel on a Mobile Device ( Poco F1 )
Trying to use "foss-billing-installation.com/admin" on a mobile device

The Main Side Panel from the Admin Panel Disappears

The Logout / Profile Information Disappears.

Only remaining Orders - Users - Invoices - Payments are Visible.

Tried Google Chrome, Mozilla, Opera All Displaying above and Hiding the Actual Admin Side Panel.

But Was Able to Use The Side Panel Using A Desktop Browser App From Playstore.

So the problem is with either the Auto Scaling to resolution or with the Admin Panlel Code Itself.

### PR DESCRIPTION
**Pull request by BelleNottelling**
Reworks our admin panel navbar to follow the original design of Tabler, and is now much more responsive. I believe this improves usability for devices of all sizes. Also closes #710
1440p:
image
1080p:
image
720p:
image
Mobile:
image

### CODE REVIEW

**Review by admdly — APPROVED** [review]
Excellent improvement in UI, imho. 👍

File: src/themes/admin_default/html/layout_default.html.twig [hunk 1]
```diff
@@ -116,91 +133,67 @@
...
+                                            </svg>
+                                        </span>
+                                        <span>{{ 'Dashboard' |trans }}</span>
+                                    </a>
+                                  </li>
+                             {% for location, group in navigation %}
+                             {% if group.subpages %}
+                                <li class="nav-item dropdown">
+                                    <a class="nav-link dropdown-toggle" href="#navbar-base" data-bs-toggle="dropdown" data-bs-auto-close="outside" role="button" aria-expanded="false" >
+                                        <svg class="icon icon-tabler icon-tabler-arrow-back-up only-large-screens">
```
Comment (evrifaessa) [inline src/themes/admin_default/html/layout_default.html.twig:158]: The icon and the text needs a little space in between.

**Review by jaapmarcus — COMMENTED** [review]
With some screen sizes the top menu doesn't go in a hamburger fast enough:
Screenshot 2023-03-14 at 11 37 34

It also mean we limit the number menu item we can handle in the top..

**Comment by BelleNottelling** [comment]
Thanks, @jaapmarcus & @evrifaessa
I've addressed your reviews, things a definitely a bit better now.

Jaap, I've changed the breakpoint for the hamburger menu to the XL size, so it'll collapse down at 1200px, which takes care of the issue for me with all possible modules enabled. And yes, this will unfortunately limit the number of menu items we can have, but I still think this is a big improvement over what's already existed, and there's some existing modules that can be moved to be a bit more sensible, such as moving the KB under the support category which will free up more space.

image
image

**Comment by John-S4** [comment]
> And yes, this will unfortunately limit the number of menu items we can have

The logical solution for that is for further menu items to be grouped under a 'More...' item, but I don't think that is in the scope of this PR really. 

The padding on the source and sponsor buttons looks very odd on the 720px screenshot, there is much more gap inside the box on the right than the left. 

Personally I would much rather not have the country code displayed in the language dropdown, because I think the flag and language name are sufficient and I don't see it adds anything. 

Apart from those little thing though, it's a massive improvement and I prefer it to any of the other proposed options with a sidebar stuck out to the left and huge weird gaps between content on bigger screens.

**Review by John-S4 — APPROVED** [review]
Approved, but also left some comments in, well, a comment. :)}

**Comment by admdly** [comment]
> Personally I would much rather not have the country code displayed in the language dropdown, because I think the flag and language name are sufficient and I don't see it adds anything.

I agree this does look a bit odd, but if it's removed then how do we handle countries with multiple official languages?

That said, there's some improvements to be made with i18n in general, at which point it could be addressed rather than in this PR.

**Comment by BelleNottelling** [comment]
> I agree this does look a bit odd, but if it's removed then how do we handle countries with multiple official languages?

The flag icons, but I don't know if they will always differ, but even then, the displayed name should be different
image

**Comment by John-S4** [comment]
> I agree this does look a bit odd, but if it's removed then how do we handle countries with multiple official languages?

Well, the language name will be different. The flag will indicate the country, and the language name the actual language. 

There are a lot of websites out there with language switchers and I'm not sure I've ever seen another one with the code included as well as flag/name.

**Comment by admdly** [comment]
> > I agree this does look a bit odd, but if it's removed then how do we handle countries with multiple official languages?
> 
> 
> 
> Well, the language name will be different. The flag will indicate the country, and the language name the actual language. 
> 
> 
> 
> There are a lot of websites out there with language switchers and I'm not sure I've ever seen many others with the code included as well as flag/name.

With English, you do often have English (US), English (UK), etc. but again, I agree this is a minor issue at this point that can be resolved with better i18n. 

The full language code is unnecessary for most languages and the country code should only be shown when clarification is needed.

**Comment by BelleNottelling** [comment]
Okay, I fixed the icon spacing John noticed. I think any changes to the locale selector can be considered out of the scope for the PR, especially since it seems like we still need to have a bit of a discussion and come to a universal agreement on it
image

**Comment by John-S4** [comment]
Fine to merge as far as I'm concerned.

**Comment by shmshd** [comment]
image

Logo is not switched between light/dark.

File: src/themes/admin_default/html/layout_default.html.twig [hunk 2]
```diff
@@ -62,17 +89,17 @@
                               <svg class="icon icon-tabler icon-tabler-arrow-back-up">
                                  <use xlink:href="#brand-github" />
                               </svg>
-                              {{ 'Source code'|trans }}
+                              <span class="only-large-screens">{{ 'Source code'|trans }}</span>
```
Comment (shmshd) [inline src/themes/admin_default/html/layout_default.html.twig:92]: I think it is best to show the text from 1024px onwards as there is quite enough space for both icon texts.

**Comment by shmshd** [comment]
image

Equal gaps?

**Review by admdly — APPROVED** [review]

**Pull request merged**